packages/core/src/server/cms/plp.ts

```diff
@@ -23,35 +27,49 @@ type PLPfromCmsEnvData = {
 
 export type PLPContentType = ContentData & PLPSettings
 
-export const getPLP = async (slug: string, previewData: Locator) => {
+export const getPLP = async (
+  slug: string,
+  previewData: Locator,
+  rewrites: Rewrite[] | RewritesConfig
+) => {
   if (config.cms.data) {
     const cmsData = JSON.parse(config.cms.data)
     const allPLPsFromCmsEnvData: PLPfromCmsEnvData[] = cmsData['plp']
 
-    return await getPLPFromCmsEnvData(`/${slug}/`, allPLPsFromCmsEnvData, {
-      ...(previewData?.contentType === 'plp' ? previewData : null),
-      contentType: 'plp',
-    })
+    return await getPLPFromCmsEnvData(
+      `/${slug}/`,
+      allPLPsFromCmsEnvData,
+      {
+        ...(previewData?.contentType === 'plp' ? previewData : null),
+        contentType: 'plp',
+      },
+      rewrites
+    )
   }
 
-  return (await getPLPFromCms(`/${slug}/`, {
-    ...(previewData?.contentType === 'plp' ? previewData : null),
-    contentType: 'plp',
-  })) as PLPContentType
+  return (await getPLPFromCms(
+    `/${slug}/`,
+    {
+      ...(previewData?.contentType === 'plp' ? previewData : null),
```

Comment:
if `contentType` was different of `plp` this code will try to destructuring a `null` and will break.

Reply:
You're right. But this change was introduced a long time ago, and it never breaks (I think). 
Can we deal with it in another PR?